src/clr-angular/forms/datepicker/date-container.spec.ts

```diff
@@ -0,0 +1,107 @@
+        beforeEach(function() {
+            TestBed.overrideComponent(ClrDateContainer, {
+                set: {
+                    providers: [
+                        {provide: DatepickerEnabledService, useClass: MockDatepickerEnabledService}, IfOpenService,
+                        DateNavigationService, LocaleHelperService, DateIOService, FormControlService
+                    ]
+                }
+            });
+
+            context = this.create(ClrDateContainer, TestComponent, []);
+
+            enabledService = <MockDatepickerEnabledService>context.getClarityProvider(DatepickerEnabledService);
+        });
+
+        describe("View Basics", () => {
+            it("renders the datepicker toggle button based on the enabled service", () => {
+                expect(enabledService.isEnabled).toBe(true);
+                expect(context.clarityElement.querySelector(".datepicker-trigger")).not.toBeNull();
+
+                enabledService.fakeIsEnabled = false;
+                context.detectChanges();
+
+                expect(context.clarityElement.querySelector(".datepicker-trigger")).toBeNull();
+            });
+
+            it("clicking on the button toggles the datepicker popover", () => {
+                spyOn(context.clarityDirective, "toggleDatepicker");
+                const button: HTMLButtonElement = context.clarityElement.querySelector(".datepicker-trigger");
+
+                button.click();
+                context.detectChanges();
+
+                expect(context.clarityDirective.toggleDatepicker).toHaveBeenCalled();
+            });
+
+            it("projects the date input", () => {
+                expect(context.clarityElement.querySelector("input")).not.toBeNull();
+            });
+
+            it("shows the datepicker view manager when .datepicker-trigger is clicked", () => {
+                expect(context.clarityElement.querySelector("clr-datepicker-view-manager")).toBeNull();
+
+                const button: HTMLButtonElement = context.clarityElement.querySelector(".datepicker-trigger");
+                button.click();
+                context.detectChanges();
+
+                expect(context.clarityElement.querySelector("clr-datepicker-view-manager")).not.toBeNull();
+            });
+        });
+
+        describe("Typescript API", () => {
+            it("toggles the datepicker popover", () => {
+                const ifOpenService: IfOpenService = context.getClarityProvider(IfOpenService);
+                const fakeEvent: MouseEvent = new MouseEvent("fakeEvent");
+                let flag: boolean;
+                const sub: Subscription = ifOpenService.openChange.subscribe((open) => {
+                    flag = open;
+                });
+
+                expect(flag).toBeUndefined();
+                context.clarityDirective.toggleDatepicker(fakeEvent);
+                context.detectChanges();
+
+                expect(flag).not.toBeUndefined();
+                expect(flag).toBe(true);
```

Comment:
The first expect is included in the second one anyway, you don't need it.

Reply:
Fixed.